src/program.ts

```diff
@@ -2856,6 +2856,15 @@ export class ClassPrototype extends Element {
     this.decoratorFlags = decoratorFlags;
   }
 
+  extends(basePtototype: ClassPrototype | null): bool {
+    if (!basePtototype) return false;
```

Comment:
Should already be covered by the first `current == basePrototype` of the loop

Reply:
right!